src/states/Preroll.js

```diff
@@ -21,10 +20,16 @@ export default class Preroll extends AdState {
    * For state transitions to work correctly, initialization should
    * happen here, not in a constructor.
    */
-  init(player, adsReady) {
+  init(player, adsReady, shouldResumeToContent) {
     // Loading spinner from now until ad start or end of ad break.
     player.addClass('vjs-ad-loading');
 
+    // If adserror, adscanceled, nopreroll or skipLinearAdMode already
+    // ocurred, resume to content immediately
+    if (shouldResumeToContent || player.ads.nopreroll_) {
```

Comment:
I realize it may be late in the game to mess with this, but is the `player.ads.nopreroll_` part still necessary? The `shouldResumeToContent` stuff seems pretty airtight.

Reply:
The main reason for it is that we set `player.ads.nopreroll_` globally after listening for the `nopreroll` event in plugin.js. My logic was that it's still possible for there to be a nopreroll event  roughly at the same time as play/one of the other negative scenarios and so we should check just in case. If we go down the route of removing `nopreroll_` from the global state and making it part of the BeforePreroll & Preroll states only, I think we would be safe to remove it then

Reply:
Yeah that sounds like the right direction to go in. Don't need to do it now.